app/src/main/java/com/cp/tasky/agenda/overview/data/repository/OverviewRepositoryImpl.kt

```diff
@@ -1,23 +1,50 @@
 package com.cp.tasky.agenda.overview.data.repository
 
+import com.cp.tasky.agenda.overview.data.mapper.AgendaOverviewMapper
 import com.cp.tasky.agenda.overview.data.remote.RemoteDataResource
 import com.cp.tasky.agenda.overview.domain.OverviewRepository
 import com.cp.tasky.agenda.overview.domain.model.Overview
-import java.time.LocalDateTime
+import com.cp.tasky.core.data.util.DataError
+import com.cp.tasky.core.data.util.Error
+import com.cp.tasky.core.data.util.Result
+import com.cp.tasky.core.data.util.map
 import javax.inject.Inject
 
 class OverviewRepositoryImpl @Inject constructor(
-    remoteDataResource: RemoteDataResource
-): OverviewRepository {
-    override suspend fun getAgendaByDate(timeStamp: LocalDateTime): Overview {
-        TODO("Not yet implemented")
+    private val remoteDataResource: RemoteDataResource,
+) : OverviewRepository {
+    //TODO: Switching from `Long` to `LocalDataTime/ZonedDataTime` once impl the DatePickerDialog
+    override suspend fun getAgendaByDate(timeStamp: Long): Result<Overview, Error> {
+        // TODO: offline first
+        return when (val agenda = remoteDataResource.getAgendaByDate(timeStamp)) {
+            is Result.Success -> {
+                agenda.map { AgendaOverviewMapper.toOverView(it) }
+            }
+
+            is Result.Error -> {
+                // TODO: fall back to local database if available
+                Result.Error(agenda.error)
+            }
+            else -> Result.Error(DataError.Remote.UNKNOWN)
+        }
     }
 
-    override suspend fun syncAgendaRemotely() {
-        TODO("Not yet implemented")
+    override suspend fun syncAgendaRemotely(): Result<Unit, Error> {
+        // TODO: get overviewSyncRequest from ROOM to sync agenda items with remote database
+        return Result.Loading // temp return value
```

Comment:
Do you need this loading value as part of the Result class? Couldn't you just use the suspend function to know when this is loading and when it isn't? 
```
isLoading = true
syncAgendaRemotely()
isLoading = false
```
Especially, because you seem to only use this loading value here for this function and not for others.